packages/react-components/react-tree/src/hooks/useRootTree.ts

```diff
@@ -25,12 +25,9 @@ export function useRootTree(
   const checkedItems = React.useMemo(() => createCheckedItems(props.checkedItems), [props.checkedItems]);
 
   const requestOpenChange = (request: Extract<TreeItemRequest, { requestType: 'open' }>) => {
-    const open = request.itemType === 'branch' && !openItems.has(request.value);
```

Comment:
Before the fix, the `open` attribute for a `leaf` node was always set to `false` but after the change it reports a "state of the leaf node":
```log
// first click
Tree::onOpenChange {value: 'leaf', open: true}

// second click
Tree::onOpenChange {value: 'leaf', open: false}
```
We already discussed this some time ago, I do not like the fact that the `onOpenChanged` is called for a leaf node which is supposed to not have any expaned/collapsed state, but I remember it is necessary for some reason?
Anyway, this is a change in behavior, are we ok with that?

Reply:
`onOpenChange` has to be called even for leaves to this behaviour here https://react.fluentui.dev/?path=/docs/components-tree--default#customizing-interaction, for this case it doesn't really matter the `open` state of the leaf as long as it calls the callback to ensure the behaviour is followed.

This is a good catch nonetheless, this would cause an unnecerssary re-render as the leaf doesn't need to update it's open state! I'll see to update and add a test to ensure this behaviour

Reply:
Can you confirm this Is resolved @miroslavstastny ?

Reply:
> I'll see to update and add a test to ensure this behaviour
Has this been done?

Reply:
yes https://github.com/microsoft/fluentui/pull/30050/files#diff-325187eb5fab0b7b6caaca9b986def282d78c7a75a3565ca875fb9de2609ad91R35